dnf/transaction_sr.py

```diff
@@ -194,21 +194,24 @@ def __init__(
         base,
         ignore_extras=False,
         ignore_installed=False,
-        skip_unavailable=False
+        skip_unavailable=False,
+        history_mode=False
     ):
         """
         :param base: the dnf base
         :param fn: the filename to load the transaction from
         :param ignore_extras: whether to ignore extra package pulled into the transaction
         :param ignore_installed: whether to ignore installed versions of packages
         :param skip_unavailable: whether to skip transaction packages that aren't available
+        :param history_mode: enable a more permissive behavior for `dnf history {undo|redo|rollback}` commands
```

Comment:
This looks a bit like a rudimentary brace. The description of the new argument doesn't say anything about its effect, just refers to it being related to history. On this level the switch should be based on its semantics and not on it's usage. The logic around this is again already non-trivial and this muddies up the waters even more.